app/src/main/java/cm/aptoide/pt/app/view/AppViewPresenter.java

```diff
@@ -118,60 +119,68 @@ public AppViewPresenter(AppViewView view, AccountNavigator accountNavigator,
     handleDonateCardImpressions();
 
     handleInterstitialAdClick();
-    handleInterstitialAdLoaded();
-    showInterstitialAd();
-  }
 
-  private Completable showBannerAd() {
-    return appViewManager.shouldLoadBannerAd()
-        .observeOn(viewScheduler)
-        .flatMapCompletable(loadBanner -> {
-          if (loadBanner) {
-            view.showBannerAd();
-          }
-          return Completable.complete();
-        });
+    loadInterstitialAd();
+    showInterstitialNew();
   }
 
-  private void showInterstitialAd() {
+  private void showInterstitialNew() {
     view.getLifecycleEvent()
         .filter(lifecycleEvent -> lifecycleEvent == View.LifecycleEvent.CREATE)
-        .flatMap(created -> view.installAppClick())
-        .flatMap(create -> appViewManager.loadAppViewViewModel()
-            .toObservable())
+        .flatMap(__ -> view.isAppViewReadyToDownload())
+        .flatMap(__ -> Observable.zip(downloadStarted(), view.interstitialAdLoaded(),
+            (downloadAppViewModel, moPubInterstitialAdClickType) -> Observable.just(
+                downloadAppViewModel)))
+        .doOnNext(__ -> view.showInterstitialAd())
+        .doOnNext(__ -> appViewAnalytics.installInterstitialImpression())
+        .observeOn(Schedulers.io())
```

Comment:
we should discuss this since I don't think the presenter should decide in which thread does the backend run, since he doesn't have enough information.